src/monmove.cpp

```diff
@@ -101,16 +101,24 @@ bool monster::is_immune_field( const field_type_id fid ) const
 bool monster::can_move_to( const tripoint &p ) const
 {
     const bool can_climb = has_flag( MF_CLIMBS ) || has_flag( MF_FLIES );
-    if( g->m.impassable( p ) && !( can_climb && g->m.has_flag( "CLIMBABLE", p ) ) ) {
+    // Using this complex check instead of generic check for DIGGABLE flag
+    // to prevent burrowing creatures appear from under certain terrain like
+    // concrete floor, doors or windows
+    const bool non_diggable = g->m.has_flag( "ROAD", p ) ||
```

Comment:
What about "t_console"? It appears to be diggable now (has neither of the flags listed here). What about "t_gas_pump"? "t_atm" is also diggable now.

Anyway: `!has_flag(DIGGABLE)` already excludes things like doors and windows and concrete floors (neither of them has the DIGGABLE flag).

Reply:
> Anyway: !has_flag(DIGGABLE) already excludes things like doors and windows and concrete floors (neither of them has the DIGGABLE flag).

AND also the chain link fence which is the reason for this PR in the first place.